frontend/lib/util/route-switch.tsx

```diff
@@ -1,15 +1,30 @@
 import React from "react";
-import { RouteComponentProps, Switch } from "react-router-dom";
+import { RouteComponentProps, Switch, Redirect } from "react-router-dom";
 import { RouteInfo } from "./route-util";
 import { NotFound } from "../pages/not-found";
 
+/**
+ * This is similar to React Router's `<Switch>` component, but
+ * additionally checks to see if the current URL pathname exists
+ * on the given `RouteInfo`.  If it does, we proceed as a normal
+ * `<Switch>` component would.
+ *
+ * However, if the route _doesn't_ seem to exist, we do one of
+ * two things:
+ *
+ *   1. If adding or removing a single `/` to the end of the pathname
+ *      results in a known route, we return a redirect.
+ *   2. Otherwise, we return a 404 page.
+ */
 export const RouteSwitch: React.FC<
   RouteComponentProps & { routes: RouteInfo<unknown, unknown> }
 > = (props) => {
   const { location } = props;
+  const { routeMap } = props.routes;
```

Comment:
Should this be `const { routeMap } = props.routes.routeMap;`?
I'm just looking at how it used to be (`!props.routes.routeMap.exists()`) and it seems we're missing one level of hierarchy here

Reply:
Oh, that's [destructuring syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) which can be a bit counterintuitive sometimes, I've made the same kind of mistake myself.  The line you suggested would actually be the equivalent of `const routeMap = props.routes.routeMap.routeMap`, which we don't want!

Reply:
Ohhh right, that makes sense. Thanks for the explainer!